lib/WebSocket.js

```diff
@@ -632,6 +634,16 @@ function initAsClient (address, protocols, options) {
 
   this._req = httpObj.get(requestOptions);
 
+  if (options.handshakeTimeout) {
+    this._req.setTimeout(options.handshakeTimeout, () => {
+      const err = new Error('opening handshake has timed out');
+      err.code = 'TIMEOUT';
```

Comment:
No problem, thought since the message is a sentence adding code could be a better way to "locate" the error, but all good.